pym/bob/scm/url.py

```diff
@@ -291,31 +291,47 @@ async def invoke(self, invoker):
 
         # Download only if necessary
         if not self.isDeterministic() or not os.path.isfile(destination):
-            try:
-                url = parseUrl(self.__url)
-            except ValueError as e:
-                invoker.fail(str(e))
-
-            if url.scheme in ['', 'file']:
-                # Verify that host name is empty or "localhost"
-                if url.netloc not in ['', 'localhost']:
-                    invoker.fail("Bad/unsupported URL: invalid host name: " + url.netloc)
-                # Local files: copy only if newer (u), target never is a directory (T)
-                if isYounger(url.path, destination):
-                    if os.path.isdir(destination):
-                        invoker.fail("Destination", destination, "is an existing directory!")
-                    invoker.trace("<cp>", url.path, workspaceFile)
-                    shutil.copy(url.path, destination)
-            elif url.scheme in ["http", "https", "ftp"]:
-                invoker.trace("<wget>", self.__url, ">", workspaceFile)
+            urlSafe = self.__url
+            urls = [self.__url] if not isinstance(self.__url, list) else self.__url
+            for index, url in enumerate(urls):
+                self.__url = url
                 try:
-                    err = await invoker.runInExecutor(UrlScm._download, self, destination)
-                    if err:
-                        invoker.fail(err)
-                except (concurrent.futures.CancelledError, concurrent.futures.process.BrokenProcessPool):
-                    invoker.fail("Download interrupted!")
-            else:
-                invoker.fail("Unsupported URL scheme: " + url.scheme)
+                    url = parseUrl(self.__url)
+                except ValueError as e:
+                    invoker.fail(str(e))
+
+                if url.scheme in ['', 'file']:
+                    # Verify that host name is empty or "localhost"
+                    if url.netloc not in ['', 'localhost']:
+                        invoker.fail("Bad/unsupported URL: invalid host name: " + url.netloc)
+                    # Local files: copy only if newer (u), target never is a directory (T)
+                    if isYounger(url.path, destination):
+                        if os.path.isdir(destination):
+                            invoker.fail("Destination", destination, "is an existing directory!")
+                        invoker.trace("<cp>", url.path, workspaceFile)
+                        shutil.copy(url.path, destination)
+                elif url.scheme in ["http", "https", "ftp"]:
+                    invoker.trace("<wget>", self.__url, ">", workspaceFile)
+                    try:
+                        err = await invoker.runInExecutor(UrlScm._download, self, destination)
```

Comment:
i tried to give that method the url, instead of self, but this hasn't worked for me for unknown reasons.
i had to use the self.__url. (ref. line 297 (new))